### PR TITLE
only run postinstall tests for updated modules

### PR DESCRIPTION
**Pull request by gurneyalex**
closes #9768

### CODE REVIEW

**Comment by gurneyalex** [comment]
upstream pr is #9773

**Comment by Garamotte** [comment]
:+1: 

Tested on some customer projects, it works as expected.
When updating module A, with module B depending on A, tests of B are run too.

**Comment by simahawk** [comment]
Just tested with and without the fix: I confirm is working.

**Comment by pedrobaeza** [comment]
:+1: 

@gurneyalex, I think this happens also on v8. Can it be?

**Comment by Garamotte** [comment]
I use this patch on all 8.0 and 9.0 projects, and it works :)

**Comment by gurneyalex** [comment]
@pedrobaeza I'll backport this to v8

**Comment by pedrobaeza** [comment]
Should we then send the patch to Odoo to v8 instead of v9?

**Comment by gurneyalex** [comment]
@pedrobaeza they are not going to merge it anyway, or maybe in 2 years...

**Comment by pedrobaeza** [comment]
hehe, OK, I merge this one anyway.

**Comment by gurneyalex** [comment]
see also #453 for the v8 version (cc @sylvain-garancher @simahawk @pedrobaeza)

File: openerp/modules/loading.py [hunk 1]
```diff
@@ -415,7 +415,10 @@ def load_modules(db, force_demo=False, status=None, update_module=False):
         t0 = time.time()
         t0_sql = openerp.sql_db.sql_counter
         if openerp.tools.config['test_enable']:
-            cr.execute("SELECT name FROM ir_module_module WHERE state='installed'")
+            if update_module:
+                cr.execute("SELECT name FROM ir_module_module WHERE state='installed' and name in %s", (tuple(mods),))
```
Comment (lmignon) [inline openerp/modules/loading.py:419]: I've a bug when tests are ran at install, sometimes mods is empty and this changes breaks my tests.. 

``` python
if update_module and mods:
    ...
```

**Comment by lmignon** [comment]
@gurneyalex @pedrobaeza This change has introduced a regression in my case https://github.com/OCA/OCB/pull/393/files#r56490157

**Comment by pedrobaeza** [comment]
Can you point me to a module/case in which happens?

**Comment by lmignon** [comment]
It's a specific module for one of our clients.

**Comment by lmignon** [comment]
``` shell
2016-03-17 11:38:07,195 16227 CRITICAL odoo-client-test openerp.service.server: Failed to initialize database `odoo-client-test`.
Traceback (most recent call last):
  File "/home/lmi/projects/openerp/odoo-client/venv/src/odoo/openerp/service/server.py", line 885, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/home/lmi/projects/openerp/odoo-client/venv/src/odoo/openerp/modules/registry.py", line 386, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/lmi/projects/openerp/odoo-client/venv/src/odoo/openerp/modules/loading.py", line 419, in load_modules
    cr.execute("SELECT name FROM ir_module_module WHERE state='installed' and name in %s", (tuple(mods),))
  File "/home/lmi/projects/openerp/odoo-client/venv/src/odoo/openerp/sql_db.py", line 141, in wrapper
    return f(self, *args, **kwargs)
  File "/home/lmi/projects/openerp/odoo-client/venv/src/odoo/openerp/sql_db.py", line 220, in execute
    res = self._obj.execute(query, params)
ProgrammingError: syntax error at or near ")"
LINE 1: ...FROM ir_module_module WHERE state='installed' and name in ()                                                                   
```

**Comment by pedrobaeza** [comment]
Do you have an alternative proposal?

**Comment by pedrobaeza** [comment]
Ah, OK, I see...

**Comment by pedrobaeza** [comment]
Making the PR for patching this one

@gurneyalex, can you modify the rest of the PRs?

**Comment by pedrobaeza** [comment]
Done in https://github.com/OCA/OCB/pull/454

**Pull request merged**